Sources/Chat/Registry.swift

```diff
@@ -3,19 +3,20 @@ import Foundation
 import WalletConnectUtils
 
 protocol Registry {
-    func register(account: Account, pubKey: String)
-    func resolve(account: Account) -> String?
+    func register(account: Account, pubKey: String) async throws
+    func resolve(account: Account) async throws -> String
 }
 
-class KeyValueRegistry: Registry  {
+actor KeyValueRegistry: Registry  {
```

Comment:
What benefits actor gives us here? 

We could make it class and make registryStore private. So it couldn't be mutated from outside too

Reply:
true it is better `registryStore` is private
but I think making it an actor gives thread safety

> So it couldn't be mutated from outside too

It could be mutated by calling register() from different threads
anyway this class is mimicking an http client right now